tests/components/awair/test_config_flow.py

```diff
@@ -1,99 +1,139 @@
 from homeassistant.core import HomeAssistant
 
-from .const import CONFIG, DEVICES_FIXTURE, NO_DEVICES_FIXTURE, UNIQUE_ID, USER_FIXTURE
+from .const import CLOUD_CONFIG, CLOUD_UNIQUE_ID, LOCAL_CONFIG, LOCAL_UNIQUE_ID
 
 from tests.common import MockConfigEntry
 
 
-async def test_show_form(hass):
+async def test_show_form(hass: HomeAssistant):
     """Test that the form is served with no input."""
     result = await hass.config_entries.flow.async_init(
         DOMAIN, context={"source": SOURCE_USER}
     )
 
-    assert result["type"] == data_entry_flow.FlowResultType.FORM
+    assert result["type"] == data_entry_flow.FlowResultType.MENU
     assert result["step_id"] == SOURCE_USER
 
 
-async def test_invalid_access_token(hass):
+async def test_invalid_access_token(hass: HomeAssistant):
     """Test that errors are shown when the access token is invalid."""
 
     with patch("python_awair.AwairClient.query", side_effect=AuthError()):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+        menu_step = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=CLOUD_CONFIG
+        )
+
+        form_step = await hass.config_entries.flow.async_configure(
+            menu_step["flow_id"],
+            {"next_step_id": "cloud"},
+        )
+
+        result = await hass.config_entries.flow.async_configure(
+            form_step["flow_id"],
+            CLOUD_CONFIG,
         )
 
         assert result["errors"] == {CONF_ACCESS_TOKEN: "invalid_access_token"}
 
 
-async def test_unexpected_api_error(hass):
+async def test_unexpected_api_error(hass: HomeAssistant):
     """Test that we abort on generic errors."""
 
     with patch("python_awair.AwairClient.query", side_effect=AwairError()):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+        menu_step = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=CLOUD_CONFIG
+        )
+
+        form_step = await hass.config_entries.flow.async_configure(
+            menu_step["flow_id"],
+            {"next_step_id": "cloud"},
+        )
+
+        result = await hass.config_entries.flow.async_configure(
+            form_step["flow_id"],
+            CLOUD_CONFIG,
         )
 
         assert result["type"] == "abort"
         assert result["reason"] == "unknown"
 
 
-async def test_duplicate_error(hass):
+async def test_duplicate_error(hass: HomeAssistant, user, cloud_devices):
     """Test that errors are shown when adding a duplicate config."""
 
     with patch(
-        "python_awair.AwairClient.query", side_effect=[USER_FIXTURE, DEVICES_FIXTURE]
+        "python_awair.AwairClient.query",
+        side_effect=[user, cloud_devices],
     ), patch(
         "homeassistant.components.awair.sensor.async_setup_entry",
```

Comment:
We only need to patch the integration setup function on create entry results or if the config entry is reloaded, eg when re-authenticating the config entry.